### ISSUE TITLE
outputDir point to the full path

### ISSUE THREAD

**Issue by maxulysse** [askimed/nf-test#66]
`outputDir` point to the full path
I'm trying to have `params.outdir` to be relative, and not really any idea on how to do so.
Any solutions to get `outputDir` to be relative?
Or any other variable that can be used for that?

This is what I have:
```
cat /home/maxime/workspace/nf-core_sarek/.nf-test/tests/6b4fc92d7235947cf7ae249c9cd9ebf/meta/params.json

{"baseDir":"/home/maxime/workspace/nf-core_sarek","outputDir":"/home/maxime/workspace/nf-core_sarek/.nf-test/tests/6b4fc92d7235947cf7ae249c9cd9ebf/output","outdir":"/home/maxime/workspace/nf-core_sarek/.nf-test/tests/6b4fc92d7235947cf7ae249c9cd9ebf/output"}
```

This is what I'm trying to do:
```
cat /home/maxime/workspace/nf-core_sarek/.nf-test/tests/6b4fc92d7235947cf7ae249c9cd9ebf/meta/params.json

{"baseDir":"/home/maxime/workspace/nf-core_sarek","outputDir":".nf-test/tests/6b4fc92d7235947cf7ae249c9cd9ebf/output","outdir":".nf-test/tests/6b4fc92d7235947cf7ae249c9cd9ebf/output"}
```
I have some files that are created with using `params.outdir` and it messes up the checksum with the full path.

**Comment by nvnieuwk** [comment]
Can you remove the `baseDir` from the `outputDir`? This would have the desired effect, right?

**Comment by maxulysse** [comment]
We could probably update [the docs](https://code.askimed.com/nf-test/testcases/global_variables/#outputdir):
This part lead me to believe the path would be relative:

> This variable points to the directory within the temporary test directory (`.nf-test/tests/<test-dir>/output/`).

**Comment by maxulysse** [comment]
BTW, solved it for now with:
```
            params {
                outdir = baseDir.toURI().relativize(outputDir.toURI()).toString()
            }
```

**Comment by maxulysse** [comment]
Hi,
Any update on that issue?
My tiny fix solves some issues for me, but not all of them

**Comment by lukfor** [comment]
Thanks 🙏 

To be sure I got it right. the following output would fit?

```
cat .nf-test/tests/ed6d3e4396651c8a5a3aca14e4dce26f/output/csv/md5sum.csv 
.nf-test/tests/ed6d3e4396651c8a5a3aca14e4dce26f/output/test_1.fastq.gz.md5
```

or did you prefer: 

```
cat .nf-test/tests/ed6d3e4396651c8a5a3aca14e4dce26f/output/csv/md5sum.csv 
output/test_1.fastq.gz.md5
```

ATM I am not sure what the best solution is: (a) update `outputDir` to relative path or (b) add a new variable with the relative path. If (a) breaks existing testcases, it could be tricky for users to find the cause.

**Comment by maxulysse** [comment]
option `a` would be good if we're sure that the path is reproducible in a different environment (let's say my computer and GHA).
option `b` would be closer to what we have in reality.
We're using a `--outdir` params in nf-core to specify the output folder, so the issue was also to get `--outdir` to point to a relative path within the nf-test framework.

**Comment by lukfor** [comment]
We are considering the following approach:

- We will leave `$outputDir` unchanged to maintain compatibility.
- Users can continue using this pre-created folder, ensuring existing tests remain unaffected.
- Instead, we will introduce `$launchDir`, which will point to the directory where the Nextflow process was initiated.
- As a result, the following variables will be available:
  - `$baseDir -> /home/maxime/workspace/nf-core_sarek` (we will also support `$projectDir` as an alias like in Nextflow)
  - `$launchDir -> /home/maxime/workspace/nf-core_sarek/.nf-test/tests/6b4fc92d7235947cf7ae249c9cd9ebf`
  - `$outputDir -> /home/maxime/workspace/nf-core_sarek/.nf-test/tests/6b4fc92d7235947cf7ae249c9cd9ebf/output`

Since the Nextflow process is now started in the directory `$launchDir`, you can set `param.outdir = "results"` in the `when` block and access the folder in the `then` closure using `path("$launchDir/results").`

Your example:

```
nextflow_workflow {

    name "Test workflow in main.nf"
    script "main.nf"
    workflow "MAIN"

    test("MAIN: default test") {
        when {
            workflow {
                """
                """

            }
            params {
                outdir = "results"
            }
        }
        then {
            assertAll(
                { assert workflow.success },
                { assert snapshot(path("$launchDir/results").list()).match("outputPath") }
            )
        }
    }
}
```

The content of the`md5sum.csv` file:

```
cat results/csv/md5sum.csv
results/test_1.fastq.gz.md5
```

 We believe that this approach and the chosen variable names are consistent with Nextflow. Happy to hear your feedback.

**Comment by lukfor** [comment]
I would also add `$moduleDir` (which points to the location of the module script) and `$moduleTestDir` (which points to the location of the test script).

Potential use case: If a module contains its own test data in the modules folder, it would be possible to load the data via:
```
inputs[0]=file("$moduleTestDir/sample.fastq.gz"). // --> modules/local/my_module/tests/sample.fastq.gz
```
or
```
inputs[0]=file("$moduleDir/sample.fastq.gz"). // --> modules/local/my_module/sample.fastq.gz
```

**Comment by maxulysse** [comment]
we usually have the test data in a separate repo, but I like where this is going

**Comment by lukfor** [comment]
We have just released RC1 of version 0.8.0 :tada:
This is a pre-release and can be installed using the following command:
```
curl -fsSL https://code.askimed.com/install/nf-test | bash -s 0.8.0-rc1
```
If everything works as expected, please feel free to close this issue. Thanks for your support! :pray:

**Comment by maxulysse** [comment]
Works as expected, thank you so much
